ci_tools/check_slots.py

```diff
@@ -92,8 +92,8 @@ def fill_dictionary(title, message, file, start_line, end_line, annotation_level
     args = parser.parse_args()
 
     # Get ast modules
-    ast_folder = os.path.dirname(ast.__file__)
-    ast_modules = [mod[:-3] for mod in os.listdir(ast_folder) if mod != '__init__.py' and mod.endswith('.py')]
+    ast_folder = pathlib.Path(ast.__file__).parent
+    ast_modules = ['.'.join(f.relative_to(ast_folder).parts)[:-3] for f in ast_folder.rglob('*.py') if f.parts[-1] != '__init__.py']
```

Comment:
Why are you taking the last three characters of a string? Should it not be this instead?
```suggestion
    ast_modules = ['.'.join(f.relative_to(ast_folder).parts[:-3]) for f in ast_folder.rglob('*.py') if f.parts[-1] != '__init__.py']
```

Reply:
I'm not taking the last three characters. I am stripping the last three characters. In other words removing the extension `.py` so I can create a module name that can be imported

Reply:
True! I read it in the opposite way! 😅